test/e2e/multihoming.go

```diff
@@ -1012,15 +1020,27 @@ var _ = Describe("Multi Homing", func() {
 					)
 					Expect(err).To(Succeed())
 
-					By("asserting the *allowed-client* pod can contact the server pod exposed endpoint")
-					Eventually(func() error {
-						return reachToServerPodFromClient(cs, serverPodConfig, allowedClientPodConfig, serverIP, port)
-					}, 2*time.Minute, 6*time.Second).Should(Succeed())
+					if permission != DenyAll {
+						By("asserting the *allowed-client* pod can contact the server pod exposed endpoint")
+						Eventually(func() error {
+							return reachToServerPodFromClient(cs, serverPodConfig, allowedClientPodConfig, serverIP, port)
+						}, 2*time.Minute, 6*time.Second).Should(Succeed())
+					} else {
```

Comment:
this is / else does not look good. Also, you're assuming "all" policies which are not deny all mean it'll be able to access the server. While (I think ...) that's true today, it doesn't mean that will be true in the future. 

Maybe consider splitting the table in two ? (one expecting successes, another expecting failures)

Also open to other alternatives.

Reply:
i can split it for sure, it will just cost code duplication
the other alternative, is to be reactive and only once the assumption you mentioned is broken to split it

whatever you prefer

Reply:
btw it is require to split to 3 (or to 2 but one of them will still have some conditions)

Reply:
what about something like 
https://github.com/ovn-org/ovn-kubernetes/pull/4156/files#diff-8a0101c1714fa2f18fcb47cc2f71b4a820bc6e88fb23f0fcb557abafbfb3b924R825

else we will need to split to 3 (deny all, allow all, and the ones that exists already today)
(or 2 maybe if two of the 2 can be combined, with some condition)